backend/src/main/kotlin/cz/loono/backend/data/model/Account.kt

```diff
@@ -0,0 +1,101 @@
+package cz.loono.backend.data.model
+
+import org.hibernate.Hibernate
+import java.time.LocalDate
+import java.util.Objects
+import javax.persistence.Column
+import javax.persistence.Embeddable
+import javax.persistence.Embedded
+import javax.persistence.Entity
+import javax.persistence.Id
+import javax.persistence.Table
+
+@Entity
+@Table(name = "\"account\"")
+data class Account(
+    @Id
+    @Column(nullable = false)
+    val uid: String = "",
+
+    @Embedded
+    val userAuxiliary: UserAuxiliary = UserAuxiliary(),
+
+    @Embedded
+    val settings: Settings = Settings(),
+
+    @Column(nullable = false)
+    val points: Int = 0,
+) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
+        other as Account
+
+        @Suppress("SENSELESS_COMPARISON")
+        return uid != null && uid == other.uid
+    }
+
+    override fun hashCode(): Int = 0
+
+    @Override
+    override fun toString(): String {
+        return this::class.simpleName + "(uid = $uid , userAuxiliary = $userAuxiliary , settings = $settings , points = $points )" // ktlint-disable max-line-length
+    }
+}
+
+@Embeddable
+data class Settings(
+    @Column(nullable = false)
+    val leaderboardAnonymizationOptIn: Boolean = true,
+
+    @Column(nullable = false)
+    val appointmentReminderEmailsOptIn: Boolean = true,
+
+    @Column(nullable = false)
+    val newsletterOptIn: Boolean = false,
+) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
+        other as Settings
+
+        return leaderboardAnonymizationOptIn == other.leaderboardAnonymizationOptIn &&
+            appointmentReminderEmailsOptIn == other.appointmentReminderEmailsOptIn &&
+            newsletterOptIn == other.newsletterOptIn
+    }
+
+    override fun hashCode(): Int =
+        Objects.hash(leaderboardAnonymizationOptIn, appointmentReminderEmailsOptIn, newsletterOptIn)
+
+    override fun toString(): String {
+        return this::class.simpleName + "(leaderboardAnonymizationOptIn = $leaderboardAnonymizationOptIn , appointmentReminderEmailsOptIn = $appointmentReminderEmailsOptIn , newsletterOptIn = $newsletterOptIn )" // ktlint-disable max-line-length
+    }
+}
+
+@Embeddable
+data class UserAuxiliary(
+    @Column(nullable = true)
+    val preferredEmail: String? = null,
+
+    @Column(nullable = true)
+    val sex: String? = null,
```

Comment:
Instead of String I would prefer SexDTO enum class or something similar.

Reply:
We can make it typed, but I strongly suggest that we don't mix Network DTOs with Database/Domain objects. If we want to have a typed sex and other enums in these models, we will duplicate the enum and then map between them. I want to keep these layers separate.

Reply:
You are right. So there will be SexDto on the network layer. And transformed from Enum to enum value as a string. String will be OK.